### PR TITLE
Parse trait aliases

### PR DESCRIPTION
**Pull request by kiljacken**
Implements the needed changes to correctly parse trait aliases.

This is my first change in the parser code, so would appreciate a comment on whether this is the right way to go about it.

### CODE REVIEW

File: crates/ra_parser/src/grammar/items/traits.rs [hunk 1]
```diff
@@ -5,19 +5,29 @@ use super::*;
 // test trait_item
 // trait T<U>: Hash + Clone where U: Copy {}
 // trait X<U: Debug + Display>: Hash + Clone where U: Copy {}
+// trait Z<U> = T<U>;
+// trait Z<U> = T<U> where U: Copy;
+// trait Z<U> = where Self: T<U>;
```
Comment (matklad) [inline crates/ra_parser/src/grammar/items/traits.rs]: It's better to move these tests directly to the `if p.eat(T![=])` condition. The main goal is *not* to test the code, the main goal is to explain which bit of code handles what

File: crates/ra_parser/src/grammar/items/traits.rs [hunk 2]
```diff
@@ -5,19 +5,29 @@ use super::*;
...
-        type_params::bounds(p);
-    }
-    type_params::opt_where_clause(p);
-    if p.at(T!['{']) {
-        trait_item_list(p);
+    if p.eat(T![=]) {
+        // trait alias
+        type_params::bounds_without_colon(p);
+        type_params::opt_where_clause(p);
+        p.expect(T![;]);
```
Comment (matklad) [inline crates/ra_parser/src/grammar/items/traits.rs]: I'd do a `return;` after this, to get rid of an `else`

File: crates/ra_parser/src/grammar/items/traits.rs (hunk 1 above)
Comment (kiljacken) [reply crates/ra_parser/src/grammar/items/traits.rs]: Ahh, smart system. I think I get it.

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**